src/CoreBundle/private/library/classes/TCMSFields/TCMSFieldText.class.php

```diff
@@ -31,34 +31,28 @@ public function GetHTML()
 
     private function renderTextArea(string $data, bool $readOnly): string
     {
-        // make usage of the textarea resizer and save scrolling by setting variable textarea size based on field content
+        $cssParts = [];
+        if ('100%' !== $this->fieldCSSwidth) {
+            $cssParts[] = 'width:'.$this->fieldCSSwidth;
+        }
 
-        if (empty($data)) {
-            $iTextareaSize = 30;
-        } elseif (strlen($data) <= 400) {
-            $iTextareaSize = 50;
-        } elseif (strlen($data) <= 1000) {
-            $iTextareaSize = 100;
-        } else {
-            $count = count(explode("\n", $data));
-            $iTextareaSize = $count * 14 + 50;
-            if ($iTextareaSize > 200) {
-                $iTextareaSize = 200;
+        if ('' !== $data) {
+            $lineCount = count(explode("\n", $data));
+            $height = min(300, 18 * ($lineCount + 2)); // 18 should correspond to the actual line height
+            if ($height > 100) {
```

Comment:
I would set a max height of 500px, because otherwise extremely long text would create a text field filling the whole page.

Reply:
There's already an upper limit of 300.

Reply:
Oh. I was blind on both eyes... Nevermind.